### PR TITLE
ci: add ci/cd with gitversion, releases and e2e test

### PR DESCRIPTION
**Pull request by LeonardHd**
adds a ci/cd process with gitversion, release creation and e2e tests.
GitVersion follows semantic versioning based on conventional commits, where types with `!` introduce a breaking change and bump major. `feat` bumps the minor version and other's bump the minor.

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Download whl
+        uses: actions/download-artifact@v3
+        with:
+          name: whl
+          path: dist
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: ./dist
+      - name: Create release
```
Comment (LeonardHd) [inline .github/workflows/cd.yml:35]: reminder: before we accept and complete this PR I need to tag the latest version we released.

**Comment by LeonardHd** [comment]
@arjendev this will enable us to version based on conventional commits. Something we would benefit from with the upcoming releases / new features.

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -4,78 +4,28 @@ name: CI-CD
...
-      - name: Upload dist
-        uses: actions/upload-artifact@v3
-        with:
-          name: dist
-          path: dist
-  publish:
-    needs: build
+  ci:
+    uses: ./.github/workflows/ci.yml
+  e2e:
```
Comment (arjendev) [inline .github/workflows/ci-cd.yml:9]: Typically in a CI/CD setup the E2E would be a step under CI (if targeting a local artifact / service?) or under CD (if targeting remote artifact / service?). Any reason you put it in it's own job?

**Review by arjendev — CHANGES_REQUESTED** [review]
Nice fundamental CICD improvements!

File: .github/workflows/ci-cd.yml (hunk 2 above)
Comment (LeonardHd) [reply .github/workflows/ci-cd.yml:9]: Mostly due to isolation of concerns and reducing the nesting level, but we can also make it part of the CI part of the pipeline 👍

**Pull request merged**